File: library/src/main/java/nl/endevelopment/r2randroid/r2rlib/models/Aircraft.java
package nl.endevelopment.r2randroid.r2rlib.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Glenn on 09-07-15.
 *
 * https://www.rome2rio.com/documentation/search#Aircraft
 * class for api v1.4
 */
public class Aircraft implements Parcelable {

    /**
     code 	        string 	Aircraft code (IATA)
     manufacturer 	string 	Manufacturer name
     model 	        string 	Model name
     */

    private String code;
    private String manufacturer;
    private String model;

    public String getCode() {
        return code;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public String getModel() {
        return model;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.code);
        dest.writeString(this.manufacturer);
        dest.writeString(this.model);
    }

    public Aircraft() {
    }

    protected Aircraft(Parcel in) {
        this.code = in.readString();
        this.manufacturer = in.readString();
        this.model = in.readString();
    }

    public static final Parcelable.Creator<Aircraft> CREATOR = new Parcelable.Creator<Aircraft>() {
        public Aircraft createFromParcel(Parcel source) {
            return new Aircraft(source);
        }

        public Aircraft[] newArray(int size) {
            return new Aircraft[size];
        }
    };
}

File: library/src/main/java/com/getroadmap/r2rlib/models/Aircraft.kt
package com.getroadmap.r2rlib.models

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Glenn on 09-07-15.
 *
 * https://www.rome2rio.com/documentation/search#Aircraft
 * class for api v1.4
 */


/**
 * code 	        string 	Aircraft code (IATA)
 * manufacturer 	string 	Manufacturer name
 * model 	        string 	Model name
 */
data class Aircraft(val code: String?,
                    val manufacturer: String?,
                    val model: String?) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(code)
        parcel.writeString(manufacturer)
        parcel.writeString(model)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Aircraft> {
        override fun createFromParcel(parcel: Parcel): Aircraft {
            return Aircraft(parcel)
        }

        override fun newArray(size: Int): Array<Aircraft?> {
            return arrayOfNulls(size)
        }
    }
}
